openmdao/devtools/problem_viewer/problem_viewer.py

```diff
@@ -180,33 +181,104 @@ def view_model(problem_or_filename, outfile='visualization', show_browser=True):
         string : The filename of the case recorder file containing the data required to build the tree.
 
     outfile : str, optional
-        The path of the output folder.  Defaults to current directory + 'visualization'.
+        The name of the final output file
 
     show_browser : bool, optional
         If True, pop up the system default web browser to view the generated html file.
         Defaults to True.
+
+    embeddable : bool, optional
+        If True, gives a single HTML file that doesn't have the <html>, <DOCTYPE>, <body>
+        and <head> tags. If False, gives a single, standalone HTML file for viewing.
     """
-    cur_dir = os.getcwd()
-    if outfile == 'visualization':
-        outfile = os.path.join(cur_dir, outfile)
+    html_begin_tags = """
+                      <html>
+                      <head>
+                        <meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
+                      </head>
+                      <body>\n
+                      """
+    html_end_tags = """
+                    </body>
+                    </html>
+                    """
 
-    model_data_filename = 'model_data.js'
-    folder_name = outfile
     code_dir = os.path.dirname(os.path.abspath(__file__))
-    model_viewer_data = _get_viewer_data(problem_or_filename)
-    model_data = json.dumps(model_viewer_data)
-
-    if os.path.isdir(folder_name):
-        shutil.rmtree(folder_name)
-
-    shutil.copytree(src=code_dir + '/visualization', dst=folder_name)
-
-    model_data_dir = os.path.join(folder_name, model_data_filename)
-    with open(model_data_dir, 'w') as f:
-        f.write('var modelData = %s' % model_data)
-
+    vis_dir = os.path.join(code_dir, "visualization")
+    libs_dir = os.path.join(vis_dir, "libs")
+    src_dir = os.path.join(vis_dir, "src")
+    style_dir = os.path.join(vis_dir, "style")
+
+    #grab the libraries
+    with open(os.path.join(libs_dir, "awesomplete.js"), "r") as f:
+        awesomplete = "%s" % (f.read())
```

Comment:
just curious why you're doing the extra step of "%s" % f.read() rather than just doing 'awesomplete = f.read()'

Reply:
Good question! I actually was just doing it to stay consistent with the original implementation. Not sure why that was originally done, but I just tried it with f.read() alone and it seems to work fine. I'll push that up.